### ISSUE TITLE
Java 11 java.sql dependency

### ISSUE THREAD

**Issue by agoeb** [cqse/teamscale-jacoco-agent#48]
Just tried running the agent with a Java 11 OpenJDK. Got the following stack trace: 

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.teamscale.jacoco.agent.PreMain.premain(PreMain.java:33)
	... 6 more
Caused by: java.lang.NoClassDefFoundError: java/sql/Time
	at shadow.com.google.gson.Gson.<init>(Gson.java:265)
	at shadow.com.google.gson.Gson.<init>(Gson.java:186)
	at shadow.retrofit2.converter.gson.GsonConverterFactory.create(GsonConverterFactory.java:42)
	at com.teamscale.client.TeamscaleServiceGenerator.createService(TeamscaleServiceGenerator.java:33)
	at com.teamscale.jacoco.agent.store.upload.teamscale.TeamscaleUploadStore.<init>(TeamscaleUploadStore.java:37)
	at com.teamscale.jacoco.agent.AgentOptions.createStore(AgentOptions.java:230)
	at com.teamscale.jacoco.agent.Agent.<init>(Agent.java:39)
	at com.teamscale.jacoco.agent.AgentOptions.createAgent(AgentOptions.java:217)
	at com.teamscale.jacoco.agent.AgentBase.premain(AgentBase.java:60)
	... 11 more
Caused by: java.lang.ClassNotFoundException: java.sql.Time
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 20 more
```

According to [Stack Overflow](https://stackoverflow.com/questions/50877484/java-sql-time-exception) we (or GSON, for that matter) must declare the dependency to java.sql in the module properties.

So it seems updating JaCoCo in #44 did not completely enable Java 11 support.

**Comment by karottenreibe** [comment]
why do we need java.sql.Time anyways? I'd expect the Service to use java.time classes, not SQL classes. Maybe this is just a messed-up auto-import?

**Comment by agoeb** [comment]
It seems GSON got its [module-info.java](https://github.com/google/gson/blob/master/gson/src/main/java/module-info.java) wrong. They [fixed it on 2018-10-18](https://github.com/google/gson/pull/1402) but have not had a release since. Should be included in 2.8.6, which is not yet available.
